warehouse/views.py

```diff
@@ -326,11 +326,41 @@ def filter_key(item):
         except ValueError:
             return 1, 0, item[0]
 
+    def form_filters_tree(split_list):
+        """
+        Takes a list of lists each of them containing filter and
+        one of it's child.
+        Returns dictionary with key being filter and value being
+        it's children
```

Comment:
Better grammar (assuming this is accurate):

Takes a list of lists, each of them containing a filter and one of its children.
Returns a dictionary, each key being a filter and each value being the filter's children.